static/js/components/Toolbar.js

```diff
@@ -39,17 +40,41 @@ export default class Toolbar extends React.Component<Props> {
   }
 
   render() {
-    const { toggleShowUserMenu, showUserMenu, profile } = this.props
+    const {
+      toggleShowUserMenu,
+      showUserMenu,
+      profile,
+      isCourseUrl
+    } = this.props
 
     return (
       <div className="navbar">
         <header className="mdc-toolbar" ref={div => (this.toolbarRoot = div)}>
           <div className="mdc-toolbar__row">
             <section className="mdc-toolbar__section mdc-toolbar__section--align-start">
-              <HamburgerAndLogo onHamburgerClick={this.toggleShowDrawer} />
+              {isCourseUrl ? (
+                <React.Fragment>
+                  <a href="http://www.mit.edu" className="mitlogo">
+                    <img
+                      src={`/static/images/${
+                        SETTINGS.use_new_branding
+                          ? "MIT_circle.svg"
+                          : "mit-logo-transparent3.svg"
+                      }`}
+                    />
+                  </a>
+                  <span className="mdc-toolbar__title">
+                    <Link to="/">MIT Open</Link>
+                    {" | "}
+                    <Link to={coursesURL()}>Courses</Link>
+                  </span>
+                </React.Fragment>
+              ) : (
+                <HamburgerAndLogo onHamburgerClick={this.toggleShowDrawer} />
```

Comment:
I think we should refactor this `Toolbar` into an HOC since the logic is starting to get complicated. These variants of the `Toolbar` (e.g. one for the course page and one for the rest of the site), would be the two concrete versions of it, similar to what we're already doing with the `Sidebar` components.

Reply:
I initially started this as a separate `CourseToolbar` component, @aliceriot [suggested modifying the existing `Toolbar` instead with a `isCourseURL` property](https://github.com/mitodl/open-discussions/pull/1913#discussion_r263889003) to determine what should be rendered, so I rewrote it that way, which does result in less code, especially taking unit test files into account, but is arguably maybe a tiny bit more complex.  Personally I'm fine with either approach, they all get the job done.  @aliceriot what do you think of an HOC toolbar with two versions?

Reply:
I'm not a huge fan of two different HOCs as a solution.

I think one way to implement this that would be cleaner than passing in the `isCourseURL` property or using two different HOCs would be to have a `<Switch>` component with routes in the relevent parts of the `Toolbar` component itself. Then the `<Route>` components there would control whether the course-specific or rest-of-the-site-specific UI is rendered, and the external API of the `Toolbar` component wouldn't need to change.

I recently refactored the channel header rendering along this sort of lines. Previously there was a `withChannelHeader` HOC that added the component to a page with an indirect method of specifying what the children should be. I instead changed it so that the `ChannelHeader` is rendered for all routes under `/c/`, and then used `<Routes>` in the `ChannelHeader`'s children to control what was rendered inside of it. I thought that was a cleaner approach that leveraged react-router more and made the code easier to follow.

Reply:
I don't see those changes involving `<Route>s` to `ChannelHeader` or it's children in master, but I gave a shot at implementing it that way as I understand it.  Seems kind of overly complex, requiring two `Switch` tags:
```python
    return (
      <div className="navbar">
        <header className="mdc-toolbar" ref={div => (this.toolbarRoot = div)}>
          <div className="mdc-toolbar__row">
            <section className="mdc-toolbar__section mdc-toolbar__section--align-start">
              <Switch>
                <Route
                  path={`${match.url}courses/`}
                  render={() => (
                    <React.Fragment>
                      <MITLogoLink />
                      <span className="mdc-toolbar__title">
                        <Link to="/">MIT Open</Link>
                        {" | "}
                        <Link to={COURSE_URL}>Courses</Link>
                      </span>
                    </React.Fragment>
                  )}
                />
                <Route
                  render={() => (
                    <HamburgerAndLogo
                      onHamburgerClick={this.toggleShowDrawer}
                    />
                  )}
                />
              </Switch>
            </section>
            <section className="mdc-toolbar__section mdc-toolbar__section--align-end user-menu-section">
              <Switch>
                <Route path={`${match.url}courses/`} />
                <Route
                  render={() =>
                    SETTINGS.allow_search ? (
                      <NavLink
                        exact
                        to={SITE_SEARCH_URL}
                        activeClassName="active"
                        className="search-link navy"
                      >
                        <i className="material-icons">search</i>
                      </NavLink>
                    ) : null
                  }
                />
              </Switch>
              <UserMenu
                toggleShowUserMenu={toggleShowUserMenu}
                showUserMenu={showUserMenu}
                profile={profile}
              />
            </section>
          </div>
        </header>
      </div>
```

At this point I'm inclined to use a hybrid of all these approaches: revert back to having two separate toolbar components (`Toolbar` and `CourseToolbar`), minus the HOC (also seems to add unnecessary complexity - still need separate toolbar components, plus the HOC), and use `Switch->Route` in `App.js` to determine which toolbar to use:

```python
        <Switch>
          <Route path={`${match.url}courses/`}>
            <CourseToolbar
              toggleShowUserMenu={this.toggleShowUserMenu}
              showUserMenu={showUserMenu}
              profile={profile}
            />
          </Route>
          <Route render={() => (
            <React.Fragment>
              <Toolbar
                toggleShowDrawer={this.toggleShowDrawer}
                toggleShowUserMenu={this.toggleShowUserMenu}
                showUserMenu={showUserMenu}
                profile={profile}
              />
              <Drawer/>
            </React.Fragment>
          )} />
```

Reply:
yeah, that seems like a reasonable approach to me

Reply:
Sounds good to me